src/ripple/app/tx/Transaction.cpp

```diff
@@ -40,13 +40,18 @@ Transaction::Transaction (STTx::ref sit, Validate validate)
         mTransactionID  = mTransaction->getTransactionID ();
         mAccountFrom    = mTransaction->getSourceAccount ();
     }
+    catch (std::exception& e)
+    {
+        reason = e.what();
+        return;
+    }
     catch (...)
     {
         return;
```

Comment:
If the intent is to swallow the exception (as it is here), I have nothing against `catch(...)`.  My problem is the `catch (...) {cleanup; throw;}` pattern.  The cleanup is better done in some local object's destructor.